clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11443,9 +11443,13 @@ def err_sycl_compiletime_property_duplication : Error<
   "can't apply %0 property twice to the same accessor">;
 def err_sycl_invalid_property_list_param_number : Error<
   "%0 must have exactly one template parameter">;
-def err_sycl_invalid_accessor_property_template_param : Error<
-  "sixth template parameter of the accessor must be of accessor_property_list "
-  "type">;
+def err_sycl_invalid_accessor_template_param_number : Error<
+  "accessor class template must have atleast five template parameters">;
+def err_sycl_invalid_accessor_template_param : Error<
+  "%select{second|third|fourth|fifth|sixth}0 "
+  "template parameter of the accessor class template must be of "
+  "%select{integer|access::mode enum|access::target enum"
+          "|access::placeholder enum|accessor_property_list}1 type">;
```

Comment:
```suggestion
  "%select{integer|'access::mode' enum|'access::target' enum"
          "|'access::placeholder' enum|'accessor_property_list'}1 type">;
```

Reply:
Also, do we want to add a `%2` for what was actually specified when there's a type mismatch?

Reply:
What do you think about something like this - 

def err_sycl_invalid_accessor_template_param : Error<
```
  "invalid parameter %2. "
  "%select{second|third|fourth|fifth|sixth}0 "
  "template parameter of the accessor class template must be of "
  "%select{integer|access::mode enum|access::target enum"
          "|access::placeholder enum|accessor_property_list}1 type">;

Reply:
I dislike, but Aaron is my word-smith here.  

Also, can we add a 'blank' option to each of the selects that just isnt' used to get rid of the madness that 0==second and 1==third?

Reply:
>I dislike, but Aaron is my word-smith here.

I dislike it too lol, which is why I posted it here for feedback before going ahead with the change. I just am not sure how else to include it.

What about something like -

  "%select{second|third|fourth|fifth|sixth}0 "
  "template parameter %2 of the accessor class must be of "
  "%select{integer|access::mode enum|access::target enum"
          "|access::placeholder enum|accessor_property_list}1 type">;

>Also, can we add a 'blank' option to each of the selects 

Can you elaborate? Do you mean just literally write "%select{blank|blank|second|third|fourth|fifth|sixth}0 "

Reply:
> Can you elaborate? Do you mean just literally write "%select{blank|blank|second|third|fourth|fifth|sixth}0 "

Nope, more like this: `"%select{||second|third|fourth|fifth|sixth}0"` where index 0 and index 1 expand to nothing.

How about:
```
def err_sycl_invalid_accessor_template_param : Error<
  "%select{|second|third|fourth|fifth|sixth}0 "
  "template parameter of the 'accessor' class template must be of "
  "%select{integer|'access::mode'|'access::target'"
          "|'access::placeholder'|'accessor_property_list'}1 type; type here is %2">;
```

Reply:
Oh I didn't know I could leave it blank that way. Thanks!

Do you prefer - 

`type here is %2 ` or  `invalid type is %2`

Reply:
I don't have a strong preference, but slightly prefer "type here is"; I usually look around to see what's most commonly done in WhateverDiagnosticKinds.td and pick something similar, so if you find a more common phrase, go for it.